spec/rspec/failure_matcher_spec.rb

```diff
@@ -316,6 +315,71 @@
           .matches?(failure)
       ).to eq(false)
     end
+
+    context 'with nested schema' do
+      class NestedContract < Dry::Validation::Contract
+        params do
+          required(:name).filled { type?(String) & min_size?(5) }
+          required(:email).filled { type?(String) }
+        end
+      end
+
+      let(:schema) do
+        Dry::Validation.Contract do
+          params do
+            required(:filters).hash(NestedContract.new.schema)
+          end
+        end
+      end
+
+      it 'succeeds when error is nested error' do
+        validation_result = schema.call(filters: {name: nil, email: 1})
+
+        failure = Failure(
+          Errors::InvalidInput.from(
+            Glia::Errors.from_dry_validation_result(validation_result),
+            errors: validation_result.errors.to_h,
+            hints: {}
+          )
+        )
+
+        expect(
+          matcher
+            .with_invalid_input
+            .containing(
+              glia_input_validation_error
+                .on(:filters, :email)
+                .with_type(Glia::Errors::INVALID_TYPE_ERROR)
+                .with_message('Email must be of type string')
+            )
+            .matches?(failure)
+        ).to eq(true)
+      end
+
+
+      it 'fails when one field message of multiple field messages does not match' do
+        validation_result = schema.call(filters: {name: nil, email: 1})
+
+        failure = Failure(
+          Errors::InvalidInput.from(
+            Glia::Errors.from_dry_validation_result(validation_result),
+            errors: validation_result.errors.to_h,
+            hints: {}
+          )
+        )
+
+        expect(
+          matcher
+            .with_invalid_input
+            .containing(
+              glia_input_validation_error
+                .on(:name).with_type(Glia::Errors::INVALID_TYPE_ERROR).with_message('Email must be of type string')
```

Comment:
shouldn't this be `.on(:filters, :name)`?

Reply:
yep